### ISSUE TITLE
Random ClassCastException when ActivityController.visible() gets called

### ISSUE THREAD

**Issue by benl-pivotal** [robolectric/robolectric#2432]
### Description

When running tests on an activity, our `setup()` is calling `Robolectric.buildActivity()`, which in turn calls `ActivityController.visible()` and will on rare occasion cause a ClassCastException with the following stacktrace:

```
    java.lang.ClassCastException: java.util.LinkedHashMap$Entry cannot be cast to java.util.HashMap$TreeNode
        at java.util.HashMap$TreeNode.moveRootToFront(HashMap.java:1819)
        at java.util.HashMap$TreeNode.treeify(HashMap.java:1936)
        at java.util.HashMap$TreeNode.split(HashMap.java:2162)
        at java.util.HashMap.resize(HashMap.java:713)
        at java.util.HashMap.putVal(HashMap.java:662)
        at java.util.HashMap.put(HashMap.java:611)
        at org.robolectric.internal.bytecode.ShadowWrangler.methodInvoked(ShadowWrangler.java:113)
        at org.robolectric.internal.bytecode.RobolectricInternals.methodInvoked(RobolectricInternals.java:23)
        at android.support.v7.internal.app.ToolbarActionBar.getMenu(ToolbarActionBar.java)
        at android.support.v7.internal.app.ToolbarActionBar.populateOptionsMenu(ToolbarActionBar.java:441)
        at android.support.v7.internal.app.ToolbarActionBar$1.run(ToolbarActionBar.java:65)
        at android.os.Handler.handleCallback(Handler.java:730)
        at android.os.Handler.dispatchMessage(Handler.java:92)
        at org.robolectric.shadows.ShadowMessageQueue.dispatchMessage(ShadowMessageQueue.java:130)
        at org.robolectric.shadows.ShadowMessageQueue.access$100(ShadowMessageQueue.java:30)
        at org.robolectric.shadows.ShadowMessageQueue$1.run(ShadowMessageQueue.java:95)
        at org.robolectric.util.Scheduler$ScheduledRunnable.run(Scheduler.java:269)
        at org.robolectric.util.Scheduler.runOneTask(Scheduler.java:179)
        at org.robolectric.util.Scheduler.advanceTo(Scheduler.java:160)
        at org.robolectric.util.Scheduler.advanceBy(Scheduler.java:143)
        at org.robolectric.util.Scheduler.unPause(Scheduler.java:54)
        at org.robolectric.shadows.ShadowLooper.unPause(ShadowLooper.java:266)
        at org.robolectric.shadows.ShadowLooper.runPaused(ShadowLooper.java:306)
        at org.robolectric.shadows.CoreShadowsAdapter$2.runPaused(CoreShadowsAdapter.java:45)
        at org.robolectric.util.ActivityController.visible(ActivityController.java:173)
...
```

##### \*Difficult to reproduce\*

Our project has ~30,000 tests. When running all tests we notice that occasionally one of the tests will fail with the stack trace above.

### Robolectric & Android Version
Robolectric 3.0
Android SDK 18

**Comment by benl-pivotal** [comment]
It looks like this was fixed in Robolectric 3.1

**Comment by benl-pivotal** [comment]
It looks like this was fixed in Robolectric 3.1
